app/controllers/users/roles_controller.rb

```diff
@@ -0,0 +1,43 @@
+class Users::RolesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_user
+  before_action :authorize_user
+  before_action :redirect_to_account_page_if_acting_on_own_user, only: %i[edit]
+
+  def edit; end
+
+  def update
+    updater = UserUpdate.new(@user, user_params, current_user, user_ip_address)
+    if updater.call
+      redirect_to edit_user_path(@user), notice: "Updated user #{@user.email} successfully"
+    else
+      render :edit
+    end
+  end
+
+private
+
+  def load_user
+    @user = User.find(params[:user_id])
+  end
+
+  def authorize_user
+    authorize(@user)
+    authorize(@user, :assign_role?)
```

Comment:
I was trying to replicate the authorization that was happening in `UsersController` as closely as possible as a first step. As I think I mentioned in a commit note, I was imagining this could be simplified, but I decided against doing anything about it in this PR, because I think there are wider problems in this area.

I would argue against just getting rid of `authorize(@user)`, because it makes assumptions about the implementations of `UserPolicy#edit?`, `UserPolicy#update?` & `UserPolicy#assign_role?`.

I did contemplate calling `UserPolicy#edit?` from `UserPolicy#assign_role?` and then only calling `authorize(@user, :assign_role?)` here, but the `UserPolicy#assign_role?` method already feels like it's diverging from the idiomatic use of Pundit and I didn't want to make that worse.

I think I'd like to leave this for now and revisit the Pundit-related functionality for all these `Users::XXX` controllers when we've split out all the relevant pages from the "edit user" page. Does that make sense?

Reply:
@chrisroos & I have just discussed this and while we think there's definitely more work to do in this area, we've agreed that it's reasonable for it to be out-of-scope for this PR.

Reply:
Yep, that makes sense to me :-)